Sources/Satori/RequestRetry/RetryInvoker.swift

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2024 The Satori Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/**
+ Listens to retry events for a particular request.
+ - Parameter retries: The number of retries made so far, including this retry.
+ - Parameter retry: A `Retry` object holding information about the retry attempt.
+ */
+public typealias RetryListener = (_ retries: Int, _ retry: Retry) -> Void
```

Comment:
Were you going to move these to a common Swift folder?

Reply:
I have refrained from using common folder as it will cause more dependency and would require changes to both satori and nakama.